### ISSUE TITLE
feat: Add a list of the richest addresses

### ISSUE THREAD

**Issue by msbrogli** [HathorNetwork/hathor-explorer#19]

**Comment by obiyankenobi** [comment]
There are 2 main decisions about the behavior of the richest addresses page we have to consider:

1) Will it update on real time? If a new tx changes the list, will it update automatically?

2) Do we have a limited number of addresses on the list (like [this](https://btc.com/stats/rich-list) - top 100 addresses)? Or should we have pagination and allow the user to go through all addresses (like [this](https://bitinfocharts.com/top-100-richest-bitcoin-addresses.html))?

In my opinion: 
1) **No** real time updates. It's very unlikely it will happen and I don't think the user has any expectations of this;

2) Have a limited number of addresses. As we're still a new crypto, I'd suggest either 10 or 20 richest addresses.

**Comment by pedroferreira1** [comment]
I agree about numbers 1 and 2. No real time updates and only a limited number of addresses. I would go with 20.

**Comment by msbrogli** [comment]
1. I think we can quickly keep a list of the K richest addresses for a given token because we already have the transactions grouped by address, more precisely O(1) on time. It would also consume O(K) of memory to keep the list in memory. If K=100, it shouldn't be a problem.

2. I agree that we must keep only the first K richest. Maybe we can set K=100, and have at most 5 pages of 20 addresses each.

3. We can add this richest addresses list in the token detail's API. But I agree that we should do it in another issue. It would also increase memory requirements.


In fact, I prefer to keep the list updated on the fly because it is quick and would not consume much memory.  Also, running an algorithm that loops through all transactions to update the richest list is O(number of transactions) and gets slower over time.

What do you think?

**Comment by obiyankenobi** [comment]
@msbrogli I'm still not sure about your idea for the first item. I was asking more regarding the behavior for the end user, not the implementation itself.

As for implementation (on the fullnode), I prefer to keep the list updated as new txs arrive. When the user makes the request, it'll be ready. That's also what you said, right?

**Comment by msbrogli** [comment]
@obiyankenobi Oh! I think I got it now. You're talking whether we should update the statistics on real time on the client, right? I agree that it is not important.

Exactly. Keep a list updated as new txs arrive.
